packages/common/http/src/request.ts

```
@@ -314,6 +327,11 @@ export class HttpRequest<T> {
         this.params = options.params;
       }
 
+      // A timeout of 0ms will get discarded.
+      if (!!options.timeout) {
```

Should we log a warning / throw an error is the timeout is negative ?

---

Then we also need to specify the behaviour of a negative value.
Currently passing a negative value will behave differently with XHR and fetch. 

XHR will ignore the parameter and fetch will abort the request immediately. I personally would go for ignoring the value.

---

If they're inconsistent I think we can throw an error.